app/routes/index.tsx

```diff
@@ -25,8 +30,72 @@ export const loader = async ({request}: Parameters<LoaderFunction>[0]) => {
   }
 }
 
+enum LoadMoreType {
+  disabled = 'disabled',
+  infini = 'infini',
+  button = 'button',
+  buttonInfini = 'buttonInfini',
+}
+
 export const shouldRevalidate: ShouldRevalidateFunction = () => false
 
+const Bottom = ({
+  isSingleQuestion,
+  fetchMore,
+}: {
+  isSingleQuestion: boolean
+  fetchMore: () => Promise<any>
+}) => {
+  const [remixSearchParams] = useSearchParams()
+  const urlLoadType = useCallback(() => {
+    const more = remixSearchParams.get('more')
+    if (more) return LoadMoreType[remixSearchParams.get('more') as keyof typeof LoadMoreType]
+  }, [remixSearchParams])
+
+  const [loadMore, setLoadMore] = useState(
+    urlLoadType() || (isSingleQuestion ? LoadMoreType.disabled : LoadMoreType.buttonInfini)
+  )
+
+  useEffect(() => {
+    const more = urlLoadType()
+    if (more) {
+      setLoadMore(more)
+    }
+  }, [setLoadMore, urlLoadType])
+
+  const buttonHandler = () => {
+    fetchMore()
```

Comment:
each button press should load more questions. This handles both basic "press for more questions" and switching to infiniscroll, as in the later case the infiniscroll might not be triggered right away on bigger screens